chef/cookbooks/ceph/recipes/mon.rb

```diff
@@ -30,9 +30,50 @@
   action :create
 end
 
+directory "/var/lib/ceph/mgr/ceph-#{mon_name}" do
+  owner "ceph"
+  group "ceph"
+  mode "0750"
+  recursive true
+  action :create
+end
+
+
 # TODO cluster name
 cluster = "ceph"
 
+
+unless File.exist?("/var/lib/ceph/mgr/ceph-#{mon_name}/done")
+  keyring = "#{Chef::Config[:file_cache_path]}/#{cluster}-#{mon_name}.mgr.keyring"
+  execute "create mgr keyring" do
+    command "ceph-authtool '#{keyring}' --create-keyring --name=mgr. --add-key='#{node["ceph"]["monitor-secret"]}' --cap mgr 'allow *'"
```

Comment:
The mgr keyring needs to be written to `/var/lib/ceph/mgr/ceph-#{mon_name}/keyring` and needs caps `mon 'allow profile mgr' osd 'allow *' mds 'allow *'`, something like the "create mds keyring" block in chef/cookbooks/ceph/recipes/mds.rb.